cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1087,15 +1088,28 @@ private boolean isValid(RunRecordDetail existing, ProgramRunStatus nextProgramSt
    *
    * @param limit count at most that many runs, stop if there are more.
    */
-  public int countActiveRuns(@Nullable Integer limit) throws IOException {
-    AtomicInteger count = new AtomicInteger(0);
+  public Map<ProgramRunStatus, Integer> countActiveRuns(@Nullable Integer limit) throws IOException {
+    Map<ProgramRunStatus, AtomicInteger> tempMap = new ConcurrentHashMap<>();
```

Comment:
Why need a concurrent hash map? Everything is happening in the local thread, right?

Reply:
Undo changes in this class.